Paco/src/com/pacoapp/paco/model/ExperimentProviderUtil.java

```diff
@@ -780,6 +811,46 @@ public void insertEvent(EventInterface eventI) {
     }
   }
 
+  public List<Event> findEventsByCriteriaQuery(String[] projection, String criteriaColumns, String criteriaValues[], String sortOrder, String limit, String groupBy, String having) {	    
+	    Cursor cursor = null;
+	    List<Event> events = null;
+	    DatabaseHelper dbHelper = new DatabaseHelper(context);
+	    if(sortOrder != null){
+	    	sortOrder = sortOrder.concat(LIMIT).concat(limit);
+	    }else{
+	    	sortOrder = LIMIT.concat(limit);
+	    }
+	    try {
+	    	List<String> allColumns = Arrays.asList(projection);
+	    	String tableIndicator = ExperimentUtil.identifyTablesInvolved(eventsOutputColumns, allColumns);
+	    	if(tableIndicator.equals(ExperimentProvider.EVENTS_OUTPUTS_TABLE_NAME)){
+	    		cursor = dbHelper.query(ExperimentProvider.EVENTS_OUTPUTS_DATATYPE, projection, criteriaColumns, criteriaValues, sortOrder, groupBy, having);
+	    	}else if(tableIndicator.equals(ExperimentProvider.EVENTS_TABLE_NAME)){
+	    		cursor = dbHelper.query(ExperimentProvider.EVENTS_DATATYPE,projection, criteriaColumns, criteriaValues, sortOrder, groupBy, having);
+	    	}else if(tableIndicator.equals(ExperimentProvider.OUTPUTS_TABLE_NAME)){
+	    		cursor = dbHelper.query(ExperimentProvider.OUTPUTS_DATATYPE, projection, criteriaColumns, criteriaValues, sortOrder, groupBy, having);
+	    	}
+	    		    	
+	    	if (cursor != null) {
+	    		events = new ArrayList<Event>();
+			    while (cursor.moveToNext()) {
+			      Event event = ExperimentUtil.createEventWithPartialResponses(cursor);
```

Comment:
Interesting scenario. Based on my understanding, 
Scenario 1: Say for example an event has 10 questions. The user wants to find a specific answer (answer='abc') for that particular event. The join table result, will have one record with the event and one response which will obviously be the answer ='abc'. I think this scenario we are good and on same page.
Scenario 2: We have an event with 10 questions. The user wants to find answers for 1 question with either/or value. So the query where clause will be like 'where answer='abc' or answer='def' '. Then the join table output record will have two or more events each where all the event record values are the same but the outputs are different. I think, this will make it easier for the client side filtering of data. This will also help preserve the ordering of the requested data. (Say the client also wants 'where answer='abc' or answer='def' order by answer desc', if we coalesce the events then we will not be able to preserve the ordering)
Let me know, if my understanding is incorrect.